### PR TITLE
Modify status protocol and afk types

### PR DESCRIPTION
**Pull request by whalemer**
As per the first part of #5590

### CODE REVIEW

**Comment by whalemer** [comment]
Client PR: zarel/pokemon-showdown-client#1317

File: PROTOCOL.md [hunk 1]
```diff
@@ -153,18 +153,18 @@ represented by a space), and the rest of the string being their username.
 `|join|USER@STATUS` or `|j|USER@STATUS`
 
 > `USER` joined the room. Optionally, `USER` can end in `@` followed by a user status message.
-> A `STATUS` starting in `!` indicates the user is away.
+> `STATUS` will start in `!` if the user is away, or '` `' otherwise.
```
Comment (Zarel) [inline PROTOCOL.md]: I don't like this change.

I think there's a sense in which it's good that a user would be able to manually set an away status by adding `!` to the beginning of a status message.

**Comment by Zarel** [comment]
I think it's a little bit premature to write this much code for a proposal that's only just gone up and I haven't taken a look at yet.

**Comment by Zarel** [comment]
Part of it's, I'm not sure if I should write my objections in your PRs or in #5590. These aren't even draft PRs!

**Comment by whalemer** [comment]
superseded by #5597

**Pull request closed without merging**